.github/workflows/docker-v2-release.yml

```diff
@@ -0,0 +1,277 @@
+name: Docker v2 release
+
+on:
+  pull_request: # TODO: Remove after testing the PR
+  release:
+    types: [published]
+  push:
+    branches:
+      - release\/*
```

Comment:
This is a change compared with `docker.yml` and isn't right.

We want to release to dockerhub only when this happens:
```
  release:
    types: [published]
```

We want to test the build (and possibly publish internally) when this happens:
```
  push:
    branches:
      - develop
      - main
      - release\/*
```